protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -1276,15 +1280,26 @@ func renderServices(services []*descriptor.Service, paths *openapiPathsObject, r
 							if err != nil {
 								return err
 							}
+							grpclog.Errorf("MessageSchema:\n%#v\n\n", messageSchema)
+							grpclog.Errorf("MessageSchema Properties:\n%#v\n\n", messageSchema.Properties)
+							grpclog.Errorf("meth.RequestType.FQMN:\n%v\n\n", meth.RequestType.FQMN())
+							grpclog.Errorf("registry:\n%v\n\n", reg)
 							if len(b.PathParams) == 0 {
 								if err := schema.setRefFromFQN(meth.RequestType.FQMN(), reg); err != nil {
 									return err
 								}
 								desc = messageSchema.Description
 							} else {
-								schema = messageSchema
-								if schema.Properties == nil || len(*schema.Properties) == 0 {
-									grpclog.Warningf("created a body with 0 properties in the message, this might be unintended: %s", *meth.RequestType)
+								if meth.Name != nil {
+									grpclog.Errorf("method name:\n%v\n\n", *meth.Name)
+									defName := fmt.Sprintf("%sBody", *meth.Name)
```

Comment:
We'll need to use more than just this, probably including the protobuf file path (with slashes replaced by `_` perhaps?).